src/components/BurgerMenu.js

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+
+export default function BurgerMenu({ links }) {
+  const ref = useRef();
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    const checkIfClickedOutside = (event) => {
```

Comment:
I would expect that `checkIfClickedOutside()` checks something and returns a boolean. Maybe something like `closeOnClickOutside()` would be better?